src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/WriteFinalDecisionAboutToSubmitHandler.java

```diff
@@ -59,43 +59,29 @@ public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callb
 
         PreSubmitCallbackResponse<SscsCaseData> preSubmitCallbackResponse = new PreSubmitCallbackResponse<>(sscsCaseData);
 
-        getDecisionNoticePointsValidationErrorMessages(sscsCaseData).forEach(preSubmitCallbackResponse::addError);
+        String benefitType = sscsCaseData.getAppeal().getBenefitType() == null ? null : sscsCaseData.getAppeal().getBenefitType().getCode();
 
-        previewDocumentService.writePreviewDocumentToSscsDocument(sscsCaseData, DRAFT_DECISION_NOTICE, sscsCaseData.getWriteFinalDecisionPreviewDocument());
-        
+        if (benefitType == null) {
+            preSubmitCallbackResponse.addError("Unexpected error - benefit type is null");
+        } else {
+
+            DecisionNoticeQuestionService questionService = decisionNoticeService.getQuestionService(benefitType);
+
+            getDecisionNoticePointsValidationErrorMessages(questionService.getPointsConditionEnumClass(), questionService, sscsCaseData)
+                .forEach(preSubmitCallbackResponse::addError);
+
+            previewDocumentService.writePreviewDocumentToSscsDocument(sscsCaseData, DRAFT_DECISION_NOTICE, sscsCaseData.getWriteFinalDecisionPreviewDocument());
+        }
         return preSubmitCallbackResponse;
     }
 
-    private List<String> getDecisionNoticePointsValidationErrorMessages(SscsCaseData sscsCaseData) {
-
-        return Arrays.stream(PointsCondition.values())
-            .filter(pointsCondition -> pointsCondition.isApplicable(sscsCaseData))
-            .map(pointsCondition ->
-                getOptionalErrorMessage(pointsCondition, sscsCaseData))
+    private <T extends PointsCondition<?>> List<String> getDecisionNoticePointsValidationErrorMessages(Class<T> enumType, DecisionNoticeQuestionService decisionNoticeQuestionService, SscsCaseData sscsCaseData) {
+        return Arrays.stream(enumType.getEnumConstants())
+            .filter(pipPointsCondition -> pipPointsCondition.isApplicable(sscsCaseData))
```

Comment:
very minor point but should `pipPointsCondition` be named something more generic?

Reply:
I've now renamed to pointsCondition